### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by Fraser999**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @brian-js

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/lib.rs [hunk 1]
```diff
@@ -29,296 +29,70 @@
...
+        unused_attributes, unused_comparisons, unused_features, unused_parens, while_true)]
 #![warn(trivial_casts, trivial_numeric_casts, unused_extern_crates, unused_import_braces,
         unused_qualifications, unused_results)]
 #![allow(box_pointers, fat_ptr_transmutes, missing_copy_implementations,
          missing_debug_implementations)]
 
+#![cfg_attr(feature="clippy", feature(plugin))]
+#![cfg_attr(feature="clippy", plugin(clippy))]
+#![cfg_attr(feature="clippy", deny(clippy))]
+#![cfg_attr(feature="clippy", deny(clippy_pedantic))]
```
Comment (ghost) [inline src/lib.rs]: The last two of these can be combined.

File: src/lib.rs [hunk 2]
```diff
@@ -29,296 +29,70 @@
...
-    }
-}
-
-/// Help function to calculate the name of a MpidHeader
-pub fn mpid_header_name(mpid_header: &MpidHeader) -> Option<XorName> {
-    match serialise(&mpid_header) {
-        Ok(encoded) => return Some(XorName(sha512::hash(&encoded[..]).0)),
-        Err(error) => {
-            error!("Failed to serialise Put request: {:?}", error);
+// Format a vector of bytes as a hexadecimal number, ellipsising all but the first and last three.
```
Comment (ghost) [inline src/lib.rs:71]: ellipsising, you've got a better dictionary than I've got.

File: src/lib.rs (hunk 1 above)
Comment (Fraser999) [inline src/lib.rs]: Nice one :)  Fixed now.

File: src/lib.rs (hunk 2 above)
Comment (Fraser999) [inline src/lib.rs:71]: Stolen from https://github.com/Fraser999/routing/blob/master/src/utils.rs#L20 unfortunately.  This should probably be a function in the maidsafe_utilities crate to be honest.

Comment (ghost) [inline src/lib.rs:71]: Which says ellipsizing!

**Pull request merged**